From cdd287c88b6af7a2a362286376209c95ec0ffcea Mon Sep 17 00:00:00 2001
From: Mitchell Hashimoto <m@mitchellh.com>
Date: Fri, 14 Feb 2025 07:23:10 -0800
Subject: [PATCH] Add back `fetch-zig-cache.sh` for packaging

See the comment on more details, which also covers when and how we can
remove this.
---
 nix/build-support/fetch-zig-cache.sh | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)
 create mode 100755 nix/build-support/fetch-zig-cache.sh

diff --git a/nix/build-support/fetch-zig-cache.sh b/nix/build-support/fetch-zig-cache.sh
new file mode 100755
index 0000000000..cb6b52a1c8
--- /dev/null
+++ b/nix/build-support/fetch-zig-cache.sh
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+# NOTE THIS IS A TEMPORARY SCRIPT TO SUPPORT PACKAGE MAINTAINERS.
+# Since #5439[1], we've been moving away from this and using an alternate
+# nix-based approach to cache our dependencies. #5733 aims to make this more
+# readily consumable by people who don't have Nix installed so that Nix
+# is not a hard dependency.
+#
+# Further, a future Zig version will hopefully fix the issue where
+# `zig build --fetch` doesn't fetch transitive dependencies[3]. When that
+# is resolved, we won't need any special machinery for the general use case
+# at all and packagers can just use `zig build --fetch`.
+#
+# [1]: https://github.com/ghostty-org/ghostty/pull/5439
+# [2]: https://github.com/ghostty-org/ghostty/pull/5733
+# [3]: https://github.com/ziglang/zig/issues/20976
+
+if [ -z ${ZIG_GLOBAL_CACHE_DIR+x} ]
+then
+  echo "must set ZIG_GLOBAL_CACHE_DIR!"
+  exit 1
+fi
+
+zig build --fetch
+zig fetch git+https://github.com/zigimg/zigimg#3a667bdb3d7f0955a5a51c8468eac83210c1439e
+zig fetch git+https://github.com/mitchellh/libxev#f6a672a78436d8efee1aa847a43a900ad773618b
